modules/standard/PCGRandom.chpl

```diff
@@ -0,0 +1,1073 @@
+  }
+
+  /*
+    Returns the next value in the random stream.
+
+    Reals are between 0 and 1; integers are in the range of the integer.
+    :returns: The next value in the random stream as type :type:`eltType`.
+   */
+  proc getNext(): eltType {
+    if parSafe then
+      PCGRandomStreamPrivate_lock$ = true;
+    const result = PCGRandomStreamPrivate_getNext_noLock();
+    if parSafe then
+      PCGRandomStreamPrivate_lock$;
+    return result;
+  }
+  /*
+    Return the next random value but within a particular range.
+   */
+  proc getNext(min: eltType, max:eltType): eltType {
+    if parSafe then
+      PCGRandomStreamPrivate_lock$ = true;
+    const result = PCGRandomStreamPrivate_getNext_noLock();
+    if parSafe then
+      PCGRandomStreamPrivate_lock$;
+    return result;
+  }
+
+
+  /*
+    Advances/rewinds the stream to the `n`-th value in the sequence.
+
+    :arg n: The position in the stream to skip to.  Must be non-negative.
+    :type n: integral
+   */
+
+  proc skipToNth(n: integral) {
+    if n <= 0 then
+      halt("PCGRandomStream.skipToNth(n) called with non-positive 'n' value", n);
+    if parSafe then
+      PCGRandomStreamPrivate_lock$ = true;
+    PCGRandomStreamPrivate_skipToNth_noLock(n);
+    if parSafe then
+      PCGRandomStreamPrivate_lock$;
+  }
+
+  /*
+    Advance/rewind the stream to the `n`-th value and return it
+    (advancing the stream by one).  This is equivalent to
+    :proc:`skipToNth()` followed by :proc:`getNext()`.
+
+    :arg n: The position in the stream to skip to.  Must be non-negative.
+    :type n: integral
+
+    :returns: The `n`-th value in the random stream as type :type:`eltType`.
+  */
+
+  proc getNth(n: integral): eltType {
+    if (n <= 0) then 
+      halt("PCGRandomStream.getNth(n) called with non-positive 'n' value", n);
+    if parSafe then
+      PCGRandomStreamPrivate_lock$ = true;
+    PCGRandomStreamPrivate_skipToNth_noLock(n);
+    const result = PCGRandomStreamPrivate_getNext_noLock();
+    if parSafe then
+      PCGRandomStreamPrivate_lock$;
+    return result;
+  }
+
+  /*
+    Fill the argument array with pseudorandom values.  This method is
+    identical to the standalone :proc:`fillRandom` procedure,
+    except that it consumes random values from the
+    :class:`PCGRandomStream` object on which it's invoked rather
+    than creating a new stream for the purpose of the call.
+
+    :arg arr: The array to be filled
+    :type arr: [] :type:`eltType`
+  */
+  proc fillRandom(arr: [] eltType) {
+    forall (x, r) in zip(arr, iterate(arr.domain, arr.eltType)) do
+      x = r;
+  }
+
+  /* Randomly shuffle a 1-D non-strided array.
+     */
+  proc shuffle(arr: [] eltType) {
+    var low = arr.domain.dim(1).low;
+    var high = arr.domain.dim(1).high;
+
+    if arr.domain.rank != 1 then
+      compilerError("Shuffle requires 1-D array");
+    //if arr.domain.type.strideable then
+    //  compilerError("Shuffle requires non-strideable 1-D array");
```

Comment:
I thought it might be useful to have shuffle/permute available on an existing RandomStream instance. However I think it would be reasonable to not do that yet... thoughts?